python_modules/dagster/dagster/_core/definitions/assets/graph/base_asset_graph.py

```diff
@@ -152,6 +152,10 @@ def owners(self) -> Sequence[str]: ...
     @abstractmethod
     def is_partitioned(self) -> bool: ...
 
+    @property
+    @abstractmethod
+    def partitions_def(self) -> Optional[PartitionsDefinition]: ...
```

Comment:
both classes that implement BaseAssetNode already implement partitions_def so i think this is fine